app/controllers/admin_controller.rb

```diff
@@ -122,30 +122,27 @@ def delete_inbox_data
   end
 
   def delete_results_data
-    @competition = competition_from_params
+    @competition = competition_from_params(associations: %i[rounds results scrambles])
 
     model = params.require(:model)
 
     if model == 'All'
       @competition.results.destroy_all
       @competition.scrambles.destroy_all
     else
-      event_id = params.require(:event_id)
-      round_type_id = params.require(:round_type_id)
+      round_id = params.require(:roundId)
 
       case model
       when Result.name
-        Result.where(competition_id: @competition.id, event_id: event_id, round_type_id: round_type_id).destroy_all
+        @competition.rounds.find(round_id).results.destroy_all
```

Comment:
You can extract this into a `round = @competition.rounds.find(round_id)` outside of the `case` block.

Reply:
Okay done